internal/engine/build_and_deployer_test.go

```diff
@@ -582,6 +582,34 @@ func TestCustomBuildWithoutFastBuild(t *testing.T) {
 	}
 }
 
+func TestCustomBuildDeterministicTag(t *testing.T) {
+	f := newBDFixture(t, k8s.EnvGKE)
+	defer f.TearDown()
+	refStr := "gcr.io/some-project-162817/sancho:deterministic-tag"
+	sha := digest.Digest("sha256:11cd0eb38bc3ceb958ffb2f9bd70be3fb317ce7d255c8a4c3f4af30e298aa1aab")
+	f.docker.Images[refStr] = types.ImageInspect{ID: string(sha)}
+
+	manifest := NewSanchoCustomBuildManifestWithTag(f, "deterministic-tag")
+	targets := buildTargets(manifest)
+
+	_, err := f.bd.BuildAndDeploy(f.ctx, f.st, targets, store.BuildStateSet{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if f.docker.BuildCount != 0 {
+		t.Errorf("Expected 0 docker build, actual: %d", f.docker.BuildCount)
+	}
+
+	if f.docker.PushCount != 1 {
```

Comment:
hmmm...afaict, this test would pass fine if the functionality in this pr was totally broken...right? please also add an assertion about the image that was pushed

Reply:
nope, it fails without the rest of the functionality in this PR. if `CustomBuild.Tag` isn't set, [we generate our own temporary tag](https://github.com/windmilleng/tilt/blob/0c6e7dc834c5cc9ecea60ac9e69118a4d2750f60/internal/build/custom_builder.go#L36) and set an env variable `$TAG=imagename:temptag`; we then [check that the expected image and expected tag were built](https://github.com/windmilleng/tilt/blob/0c6e7dc834c5cc9ecea60ac9e69118a4d2750f60/internal/build/custom_builder.go#L66).

if i remove the changes to `CustomBuilder.Build`, test fails with:
```
Custom Build: Injecting Environment Variables
TAG=gcr.io/some-project-162817/sancho:tilt-build-1551202573
Running custom build cmd "true"

ERROR: fake docker client error: object not found (fakeClient.Images key: gcr.io/some-project-162817/sancho:tilt-build-1551202573)
```
(i.e. we spoof that the custom build command built `gcr.io/.../sancho:deterministic-tag`; but Build goes looking for `sancho:tilt-build-1551202573`, cant find it, thinks something has gone wrong.)